pkg/prometheus/promcfg_test.go

```diff
@@ -8837,3 +8889,87 @@ scrape_configs:
 
 	}
 }
+
+func TestTracingConfig(t *testing.T) {
+	boolFalse := false
+	testCases := []struct {
+		tracingConfig  *monitoringv1.PrometheusTracingConfig
+		name           string
+		expectedConfig string
+		expectedErr    bool
+	}{
+		{
+			tracingConfig: &monitoringv1.PrometheusTracingConfig{
+				Endpoint: "test.com:80",
+				// Expect to be a float between 0 and 1
+				SamplingFraction: "2",
+			},
+			name:           "Invalid sampling fraction",
+			expectedConfig: "",
+			expectedErr:    true,
+		},
+		{
+			tracingConfig: &monitoringv1.PrometheusTracingConfig{
+				ClientType:       "grpc",
+				Endpoint:         "https://otel-collector.default.svc.local:3333",
+				SamplingFraction: "0.5",
+				Headers: map[string]string{
+					"custom": "header",
+				},
+				Compression: "gzip",
+				Timeout:     "10s",
+				Insecure:    &boolFalse,
+			},
+			name:        "Expect valid config",
+			expectedErr: false,
+			expectedConfig: `global:
+  evaluation_interval: 30s
+  scrape_interval: 30s
+  external_labels:
+    prometheus: default/test
+    prometheus_replica: $(POD_NAME)
+scrape_configs: []
+tracing:
+  client_type: grpc
+  endpoint: https://otel-collector.default.svc.local:3333
+  sampling_fraction: 0.5
+  insecure: false
+  headers:
+    custom: header
+  compression: gzip
+  timeout: 10s
+`,
+		},
+	}
+	for _, tc := range testCases {
+		p := defaultPrometheus()
```

Comment:
Place test cases inside of `t.Run` which will make it easier to identify which test fails
```suggestion
	for _, tc := range testCases {
		t.Run(tc.name, func(t *testing.T) {
			p := defaultPrometheus()
			.....
		}
	}
```